File: Momento/History.cs
//to manage state


namespace Momento
{
    public class History
    {
        private readonly IList<EditorState> _states = new List<EditorState>();

        public void Push(EditorState state)
        {
            _states.Add(state);
        }

        //retreive last state and delete it after that
        public EditorState Pop()
        {
            var lastIndex = _states.Count - 1;
            var lastState = _states[lastIndex];
            _states.RemoveAt(lastIndex);
            return lastState;
        }
    }
}

File: DesignPattern/src/Momento/History.java
package Momento;

import java.util.ArrayList;
import java.util.List;

public class History {
    private List<EditorState> states = new ArrayList<>();

    public void Push(EditorState state)
    {
        states.add(state);
    }

    public EditorState Pop()
    {
        var lastIndex = states.size() - 1;
        var lastState = states.get(lastIndex);
        states.remove(lastIndex);

        return lastState;
    }
}
